### PR TITLE
feat: txmanager, feat: hb

### PR DESCRIPTION
**Pull request by zeroFruit**
## Related Issue

resolve: #*(Write down your issue number)*

## Description

- implement tx manager
- implement honeybadger

## Checklist

Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] Applying goimports
- [x] Test case
- [x] End of Work

### CODE REVIEW

File: queue.go [hunk 1]
```diff
@@ -95,47 +97,113 @@ func (q *MemTxQueue) Push(tx Transaction) {
...
+		if !m.hb.OnConsensus() {
+			m.tryPropose()
+		}
+		time.Sleep(m.tryInterval)
+	}
+}
+
+// tryPropose create contribution and send it to honeybadger only when
+// transaction queue size is larger than batch size
+func (m *DefaultTxQueueManager) tryPropose() error {
```
Comment (AgwaB) [inline queue.go:187]: 1. HBBFT's contribution size is `Batch/N`.
2. I think we didn't need to set tx count as `Batch/N`. it is maximum count of transactions which will be agreed.

File: queue.go [hunk 2]
```diff
@@ -95,47 +97,113 @@ func (q *MemTxQueue) Push(tx Transaction) {
...
+		return err
+	}
+	return m.hb.HandleContribution(contribution)
+}
+
 // Create batch polling random transaction in queue
-func (m *DefaultTxQueueManager) createBatch(n, b int) (Batch, error) {
-	candidate, err := m.loadCandidateTx(min(b, m.txQueue.Len()))
+func (m *DefaultTxQueueManager) createContribution() (Contribution, error) {
+	candidate, err := m.loadCandidateTx(min(m.blockSize, m.txQueue.Len()))
```
Comment (AgwaB) [inline queue.go]: what's purpose of `min(m.blockSize, m.txQueue.Len())`?

File: test/mock/honeybadger.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package mock
+
+import (
+	"github.com/DE-labtory/cleisthenes"
+	"github.com/DE-labtory/cleisthenes/pb"
+)
+
+type HoneyBadger struct {
```
Comment (AgwaB) [inline test/mock/honeybadger.go]: What is this package for?
`queue_internal_test.go` use internal `MockHoneyBadger`

**Review by AgwaB — COMMENTED** [review]
And in `queue.go`, in `selectRandomTx` method, you intent to order of logic as "after return batch, remove items" ?
```go
return batch, func() {
		for _, item := range candidate {
			m.txQueue.Push(item)
		}
	}
```

File: core/core.go [hunk 4]
```diff
@@ -116,7 +142,7 @@ func (n *Node) connect(addr cleisthenes.Address) error {
 	}()
 
 	n.connPool.Add(addr, conn)
-	n.memberMap.Add(&cleisthenes.Member{Address: addr})
+	n.memberMap.Add(cleisthenes.NewMemberWithAddress(addr))
 	return nil
 }
```
Comment (AgwaB) [inline core/core.go]: Maybe we need `ConnectAll` and `Connect` for dynamic network?

**Comment by AgwaB** [comment]
We need `network` package?

File: queue.go (hunk 1 above)
Comment (zeroFruit) [reply queue.go:187]: yes, we now define **batch** as a result of consensus, now I'll fix it as **contribution**

File: test/mock/honeybadger.go (hunk 3 above)
Comment (zeroFruit) [reply test/mock/honeybadger.go]: not used. I'll delete it!

File: queue.go (hunk 2 above)
Comment (zeroFruit) [reply queue.go]: in this step, we pick up candidates as much as block size and this is max value. if transaction queue has not that size of transaction, pick all of the transaction from the queue

**Review by AgwaB — APPROVED** [review]

**Pull request merged**